lib/internal/Magento/Framework/Data/Collection/AbstractDb.php

```diff
@@ -217,7 +218,7 @@ public function getSize()
     {
         if ($this->_totalRecords === null) {
             $sql = $this->getSelectCountSql();
-            $this->_totalRecords = $this->getConnection()->fetchOne($sql, $this->_bindParams);
+            $this->_totalRecords = $this->_totalRecords ?? $this->getConnection()->fetchOne($sql, $this->_bindParams);
```

Comment:
It seems that when if reach this point, $this->_totalRecords must be strictly null at the time null coalescence check. Or am i missing something?

@orlangur @Den4ik

Reply:
@korostii no. Value can be set during getSelectCountSql method execution, and Elasticsearch adapter does it at some point.